src/backend/distributed/planner/multi_logical_planner.c

```diff
@@ -515,6 +545,56 @@ HasUnsupportedJoinWalker(Node *node, void *context)
 
 
 /*
+ * ErrorHintRequired returns true if error hint shold be displayed with the
+ * query error message. Error hint is valid only for queries involving reference
+ * and hash partitioned tables. If more than one hash distributed table is
+ * present we display the hint only if the tables are colocated. If the query
+ * only has reference table(s), then it is handled by router planner.
+ */
+static bool
+ErrorHintRequired(char *errorHint, Query *queryTree)
+{
+	List *rangeTableList = NIL;
+	ListCell *rangeTableCell = NULL;
+	List *colocationIdList = NIL;
+
+	if (errorHint == NULL)
+	{
+		return false;
+	}
+
+	ExtractRangeTableRelationWalker((Node *) queryTree, &rangeTableList);
+	foreach(rangeTableCell, rangeTableList)
+	{
+		RangeTblEntry *rte = (RangeTblEntry *) lfirst(rangeTableCell);
+		Oid relationId = rte->relid;
+		char partitionMethod = PartitionMethod(relationId);
+		if (partitionMethod == DISTRIBUTE_BY_NONE)
```

Comment:
We discussed this for some time in some different places such as [1](https://github.com/citusdata/citus/issues/879#issuecomment-264102731) and [2](https://github.com/citusdata/citus/pull/1018#discussion_r91264511), but couldn't find a better alternative. Even, initially it was `DISTRIBUTE_BY_ALL` in the PR :) 

The behavior is the following:
* Single shard, replicated to all nodes
* Distribution column is NULL
* Shard min/max values are NULL
* All modifications goes through 2PC to prevent any invalid placements [not checked-in yet, PR is open]

And [this](https://github.com/citusdata/citus/blob/master/src/test/regress/expected/multi_reference_table.out#L19) regression test reveals it nicely.

Initially, I was also thinking that `DISTRIBUTE_BY_NONE` is not the best option. But, considering the behavior mentioned above, I became more satisfied with the name. Finally, since this is not something user facing, I decided to go on with a *reasonable* name that @ozgune is OK and @marcocitus  doesn't reject.